src/sentry/rules/conditions/event_frequency.py

```diff
@@ -208,28 +210,76 @@ def batch_query_hook(
 
-    def get_rate(self, event: GroupEvent, interval: str, environment_id: int) -> int:
-        _, duration = self.intervals[interval]
-        end = timezone.now()
+    def get_option_override(self, duration: timedelta):
         # For conditions with interval >= 1 hour we don't need to worry about read your writes
         # consistency. Disable it so that we can scale to more nodes.
         option_override_cm: contextlib.AbstractContextManager[object] = contextlib.nullcontext()
         if duration >= timedelta(hours=1):
             option_override_cm = options_override({"consistent": False})
+        return option_override_cm
+
+    def get_start_end_from_duration(self, duration: timedelta):
+        end = timezone.now()
+        start = end - duration
+        return (start, end)
+
+    def get_comparison_start_end(self, interval: timedelta, duration: timedelta):
+        end = timezone.now() - interval
+        start = end - duration
+        return (start, end)
+
+    def get_rate(
+        self,
+        duration: timedelta,
+        interval: timedelta,
+        event: GroupEvent,
+        environment_id: int,
+        comparison_type: str,
+    ) -> int:
+        start, end = self.get_start_end_from_duration(duration)
+        option_override_cm = self.get_option_override(duration)
         with option_override_cm:
-            result: int = self.query(event, end - duration, end, environment_id=environment_id)
-            comparison_type = self.get_option("comparisonType", ComparisonType.COUNT)
-            if comparison_type == ComparisonType.PERCENT:
-                comparison_interval = COMPARISON_INTERVALS[self.get_option("comparisonInterval")][1]
-                comparison_end = end - comparison_interval
-                # TODO: Figure out if there's a way we can do this less frequently. All queries are
-                # automatically cached for 10s. We could consider trying to cache this and the main
-                # query for 20s to reduce the load.
-                comparison_result = self.query(
-                    event, comparison_end - duration, comparison_end, environment_id=environment_id
-                )
-                result = percent_increase(result, comparison_result)
+            result = self.query(event, start, end, environment_id=environment_id)
+        if comparison_type == ComparisonType.PERCENT:
+            # TODO: Figure out if there's a way we can do this less frequently. All queries are
+            # automatically cached for 10s. We could consider trying to cache this and the main
+            # query for 20s to reduce the load.
+            start, comparison_end = self.get_comparison_start_end(interval, duration)
+            comparison_result = self.query(
+                event, start, comparison_end, environment_id=environment_id
+            )
+            result = percent_increase(result, comparison_result)
+        return result
 
+    def get_rate_bulk(
```

Comment:
This is untested but also unused, the entire point of refactoring this is so I can use it in https://github.com/getsentry/sentry/pull/69167 which does / will have tests for this path.

Reply:
Should we change the merge target to be `ceorourke/apply_delayed_rules` instead of `master` then? it might limit any risk of changing `get_rate` which is actively being used.

Reply:
If this is untested and unused, can we simply move it into #69167 instead?